tpu-client-next/src/quic_networking.rs

```diff
@@ -47,10 +48,23 @@ pub(crate) fn create_client_config(client_certificate: QuicClientCertificate) ->
 }
 
 pub(crate) fn create_client_endpoint(
-    bind_addr: SocketAddr,
+    bind: BindTarget,
     client_config: ClientConfig,
 ) -> Result<Endpoint, QuicError> {
-    let mut endpoint = Endpoint::client(bind_addr).map_err(IoErrorWithPartialEq::from)?;
+    let mut endpoint = match bind {
+        BindTarget::Address(bind_addr) => {
+            Endpoint::client(bind_addr).map_err(IoErrorWithPartialEq::from)?
```

Comment:
Don't really like here these `map_err` but haven't found a way to simplify this code without introducing new functions.